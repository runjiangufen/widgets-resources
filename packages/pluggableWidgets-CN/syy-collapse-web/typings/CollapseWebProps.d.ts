/**
 * This file was generated from CollapseWeb.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix UI Content Team
 */
import { ComponentType, CSSProperties, ReactNode } from "react";
import { ActionValue } from "mendix";

export type CollapsibleEnum = "header" | "disabled";

export type ExpandIconPositionEnum = "start" | "end";

export interface CollapseWebContainerProps {
    name: string;
    class: string;
    style?: CSSProperties;
    tabIndex?: number;
    children?: ReactNode;
    accordion: boolean;
    bordered: boolean;
    collapsible: CollapsibleEnum;
    expandIconPosition: ExpandIconPositionEnum;
    ghost: boolean;
    onChangeAction?: ActionValue;
}

export interface CollapseWebPreviewProps {
    className: string;
    style: string;
    styleObject?: CSSProperties;
    readOnly: boolean;
    children: { widgetCount: number; renderer: ComponentType<{ caption?: string }> };
    accordion: boolean;
    bordered: boolean;
    collapsible: CollapsibleEnum;
    expandIconPosition: ExpandIconPositionEnum;
    ghost: boolean;
    onChangeAction: {} | null;
}
