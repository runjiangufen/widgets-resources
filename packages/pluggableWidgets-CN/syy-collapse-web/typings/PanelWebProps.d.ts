/**
 * This file was generated from PanelWeb.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix UI Content Team
 */
import { ComponentType, CSSProperties, ReactNode } from "react";

export type CollapsibleEnum = "header" | "disabled" | "undefined";

export interface PanelWebContainerProps {
    name: string;
    class: string;
    style?: CSSProperties;
    tabIndex?: number;
    children?: ReactNode;
    key: string;
    collapsible: CollapsibleEnum;
    extra?: ReactNode;
    forceRender: boolean;
    header: ReactNode;
    showArrow: boolean;
}

export interface PanelWebPreviewProps {
    className: string;
    style: string;
    styleObject?: CSSProperties;
    readOnly: boolean;
    children: { widgetCount: number; renderer: ComponentType<{ caption?: string }> };
    key: string;
    collapsible: CollapsibleEnum;
    extra: { widgetCount: number; renderer: ComponentType<{ caption?: string }> };
    forceRender: boolean;
    header: { widgetCount: number; renderer: ComponentType<{ caption?: string }> };
    showArrow: boolean;
}
