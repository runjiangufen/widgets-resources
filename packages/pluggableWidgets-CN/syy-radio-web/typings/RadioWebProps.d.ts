/**
 * This file was generated from RadioWeb.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix UI Content Team
 */
import { ActionValue, DynamicValue, EditableValue, ListValue, ListAttributeValue } from "mendix";
import { Big } from "big.js";

export type DatasTypeEnum = "dynamic" | "static";

export interface StaticDataType {
    manualValue: string;
    manuaTitle: string;
    manualDisabled: boolean;
}

export type SizeEnum = "large" | "middle" | "small";

export type ButtonStyleEnum = "outline" | "solid";

export type OptionTypeEnum = "default" | "button";

export type DirectionEnum = "horizontal" | "vertical";

export interface StaticDataPreviewType {
    manualValue: string;
    manuaTitle: string;
    manualDisabled: boolean;
}

export interface RadioWebContainerProps {
    name: string;
    tabIndex?: number;
    id: string;
    datasType: DatasTypeEnum;
    staticData: StaticDataType[];
    data?: ListValue;
    attribute?: ListAttributeValue<string | Date | Big>;
    title?: ListAttributeValue<string>;
    attributedata?: EditableValue<string>;
    selectedValue?: EditableValue<string | Date | Big>;
    isText: boolean;
    size: SizeEnum;
    buttonStyle: ButtonStyleEnum;
    disabled: DynamicValue<boolean>;
    radioName: string;
    optionType: OptionTypeEnum;
    direction: DirectionEnum;
    onChangeAction?: ActionValue;
    customWidgetOnChangeValidate: boolean;
}

export interface RadioWebPreviewProps {
    readOnly: boolean;
    datasType: DatasTypeEnum;
    staticData: StaticDataPreviewType[];
    data: {} | { type: string } | null;
    attribute: string;
    title: string;
    attributedata: string;
    selectedValue: string;
    isText: boolean;
    size: SizeEnum;
    buttonStyle: ButtonStyleEnum;
    disabled: string;
    radioName: string;
    optionType: OptionTypeEnum;
    direction: DirectionEnum;
    onChangeAction: {} | null;
    customWidgetOnChangeValidate: boolean;
}
