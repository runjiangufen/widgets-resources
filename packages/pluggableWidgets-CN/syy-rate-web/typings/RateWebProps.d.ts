/**
 * This file was generated from RateWeb.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix UI Content Team
 */
import { ActionValue, DynamicValue, EditableValue } from "mendix";
import { Big } from "big.js";

export interface RateWebContainerProps {
    name: string;
    tabIndex?: number;
    id: string;
    value: EditableValue<Big>;
    allowClear: boolean;
    allowHalf: boolean;
    count: number;
    disabled: DynamicValue<boolean>;
    onChange?: ActionValue;
    customWidgetOnChangeValidate: boolean;
}

export interface RateWebPreviewProps {
    readOnly: boolean;
    value: string;
    allowClear: boolean;
    allowHalf: boolean;
    count: number | null;
    disabled: string;
    onChange: {} | null;
    customWidgetOnChangeValidate: boolean;
}
