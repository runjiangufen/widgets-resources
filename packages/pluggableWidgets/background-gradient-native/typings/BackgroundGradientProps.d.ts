/**
 * This file was generated from BackgroundGradient.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix UI Content Team
 */
import { ComponentType, CSSProperties, ReactNode } from "react";
import { ActionValue } from "mendix";
import { Big } from "big.js";

export interface ColorListType {
    color: string;
    offset: Big;
}

export interface ColorListPreviewType {
    color: string;
    offset: number | null;
}

export interface BackgroundGradientProps<Style> {
    name: string;
    style: Style[];
    colorList: ColorListType[];
    content: ReactNode;
    onClick?: ActionValue;
}

export interface BackgroundGradientPreviewProps {
    className: string;
    style: string;
    styleObject?: CSSProperties;
    readOnly: boolean;
    colorList: ColorListPreviewType[];
    content: { widgetCount: number; renderer: ComponentType<{ caption?: string }> };
    onClick: {} | null;
}
