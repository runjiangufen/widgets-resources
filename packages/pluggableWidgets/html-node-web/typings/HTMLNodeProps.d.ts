/**
 * This file was generated from HTMLNode.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix UI Content Team
 */
import { ComponentType, CSSProperties, ReactNode } from "react";
import { ActionValue, DynamicValue, ListValue, ListActionValue, ListExpressionValue, ListWidgetValue } from "mendix";

export type TagNameEnum =
    | "div"
    | "span"
    | "p"
    | "ul"
    | "ol"
    | "li"
    | "a"
    | "img"
    | "h1"
    | "h2"
    | "h3"
    | "h4"
    | "h5"
    | "h6"
    | "__customTag__";

export type AttributeValueTypeEnum = "expression" | "template";

export interface AttributesType {
    attributeName: string;
    attributeValueType: AttributeValueTypeEnum;
    attributeValueTemplate?: DynamicValue<string>;
    attributeValueExpression?: DynamicValue<string>;
    attributeValueTemplateRepeat?: ListExpressionValue<string>;
    attributeValueExpressionRepeat?: ListExpressionValue<string>;
}

export type TagContentModeEnum = "container" | "innerHTML";

export type EventNameEnum =
    | "onClick"
    | "onChange"
    | "onFocus"
    | "onLeave"
    | "onSubmit"
    | "onKeyDown"
    | "onKeyUp"
    | "onCopy"
    | "onCopyCapture"
    | "onCut"
    | "onCutCapture"
    | "onPaste"
    | "onPasteCapture"
    | "onCompositionEnd"
    | "onCompositionEndCapture"
    | "onCompositionStart"
    | "onCompositionStartCapture"
    | "onCompositionUpdate"
    | "onCompositionUpdateCapture"
    | "onFocusCapture"
    | "onBlur"
    | "onBlurCapture"
    | "onChangeCapture"
    | "onBeforeInput"
    | "onBeforeInputCapture"
    | "onInput"
    | "onInputCapture"
    | "onReset"
    | "onResetCapture"
    | "onSubmitCapture"
    | "onInvalid"
    | "onInvalidCapture"
    | "onLoad"
    | "onLoadCapture"
    | "onError"
    | "onErrorCapture"
    | "onKeyDownCapture"
    | "onKeyPress"
    | "onKeyPressCapture"
    | "onKeyUpCapture"
    | "onAbort"
    | "onAbortCapture"
    | "onCanPlay"
    | "onCanPlayCapture"
    | "onCanPlayThrough"
    | "onCanPlayThroughCapture"
    | "onDurationChange"
    | "onDurationChangeCapture"
    | "onEmptied"
    | "onEmptiedCapture"
    | "onEncrypted"
    | "onEncryptedCapture"
    | "onEnded"
    | "onEndedCapture"
    | "onLoadedData"
    | "onLoadedDataCapture"
    | "onLoadedMetadata"
    | "onLoadedMetadataCapture"
    | "onLoadStart"
    | "onLoadStartCapture"
    | "onPause"
    | "onPauseCapture"
    | "onPlay"
    | "onPlayCapture"
    | "onPlaying"
    | "onPlayingCapture"
    | "onProgress"
    | "onProgressCapture"
    | "onRateChange"
    | "onRateChangeCapture"
    | "onSeeked"
    | "onSeekedCapture"
    | "onSeeking"
    | "onSeekingCapture"
    | "onStalled"
    | "onStalledCapture"
    | "onSuspend"
    | "onSuspendCapture"
    | "onTimeUpdate"
    | "onTimeUpdateCapture"
    | "onVolumeChange"
    | "onVolumeChangeCapture"
    | "onWaiting"
    | "onWaitingCapture"
    | "onAuxClick"
    | "onAuxClickCapture"
    | "onClickCapture"
    | "onContextMenu"
    | "onContextMenuCapture"
    | "onDoubleClick"
    | "onDoubleClickCapture"
    | "onDrag"
    | "onDragCapture"
    | "onDragEnd"
    | "onDragEndCapture"
    | "onDragEnter"
    | "onDragEnterCapture"
    | "onDragExit"
    | "onDragExitCapture"
    | "onDragLeave"
    | "onDragLeaveCapture"
    | "onDragOver"
    | "onDragOverCapture"
    | "onDragStart"
    | "onDragStartCapture"
    | "onDrop"
    | "onDropCapture"
    | "onMouseDown"
    | "onMouseDownCapture"
    | "onMouseEnter"
    | "onMouseLeave"
    | "onMouseMove"
    | "onMouseMoveCapture"
    | "onMouseOut"
    | "onMouseOutCapture"
    | "onMouseOver"
    | "onMouseOverCapture"
    | "onMouseUp"
    | "onMouseUpCapture"
    | "onSelect"
    | "onSelectCapture"
    | "onTouchCancel"
    | "onTouchCancelCapture"
    | "onTouchEnd"
    | "onTouchEndCapture"
    | "onTouchMove"
    | "onTouchMoveCapture"
    | "onTouchStart"
    | "onTouchStartCapture"
    | "onPointerDown"
    | "onPointerDownCapture"
    | "onPointerMove"
    | "onPointerMoveCapture"
    | "onPointerUp"
    | "onPointerUpCapture"
    | "onPointerCancel"
    | "onPointerCancelCapture"
    | "onPointerEnter"
    | "onPointerEnterCapture"
    | "onPointerLeave"
    | "onPointerLeaveCapture"
    | "onPointerOver"
    | "onPointerOverCapture"
    | "onPointerOut"
    | "onPointerOutCapture"
    | "onGotPointerCapture"
    | "onGotPointerCaptureCapture"
    | "onLostPointerCapture"
    | "onLostPointerCaptureCapture"
    | "onScroll"
    | "onScrollCapture"
    | "onWheel"
    | "onWheelCapture"
    | "onAnimationStart"
    | "onAnimationStartCapture"
    | "onAnimationEnd"
    | "onAnimationEndCapture"
    | "onAnimationIteration"
    | "onAnimationIterationCapture"
    | "onTransitionEnd"
    | "onTransitionEndCapture";

export interface EventsType {
    eventName: EventNameEnum;
    eventAction?: ActionValue;
    eventActionRepeat?: ListActionValue;
    eventStopPropagation: boolean;
    eventPreventDefault: boolean;
}

export interface AttributesPreviewType {
    attributeName: string;
    attributeValueType: AttributeValueTypeEnum;
    attributeValueTemplate: string;
    attributeValueExpression: string;
    attributeValueTemplateRepeat: string;
    attributeValueExpressionRepeat: string;
}

export interface EventsPreviewType {
    eventName: EventNameEnum;
    eventAction: {} | null;
    eventActionRepeat: {} | null;
    eventStopPropagation: boolean;
    eventPreventDefault: boolean;
}

export interface HTMLNodeContainerProps {
    name: string;
    class: string;
    style?: CSSProperties;
    tabIndex?: number;
    tagName: TagNameEnum;
    tagNameCustom: string;
    attributes: AttributesType[];
    tagUseRepeat: boolean;
    tagContentRepeatDataSource?: ListValue;
    tagContentMode: TagContentModeEnum;
    tagContentHTML?: DynamicValue<string>;
    tagContentContainer?: ReactNode;
    tagContentRepeatHTML?: ListExpressionValue<string>;
    tagContentRepeatContainer?: ListWidgetValue;
    events: EventsType[];
}

export interface HTMLNodePreviewProps {
    className: string;
    style: string;
    styleObject?: CSSProperties;
    readOnly: boolean;
    tagName: TagNameEnum;
    tagNameCustom: string;
    attributes: AttributesPreviewType[];
    tagUseRepeat: boolean;
    tagContentRepeatDataSource: {} | { type: string } | null;
    tagContentMode: TagContentModeEnum;
    tagContentHTML: string;
    tagContentContainer: { widgetCount: number; renderer: ComponentType<{ caption?: string }> };
    tagContentRepeatHTML: string;
    tagContentRepeatContainer: { widgetCount: number; renderer: ComponentType<{ caption?: string }> };
    events: EventsPreviewType[];
}
