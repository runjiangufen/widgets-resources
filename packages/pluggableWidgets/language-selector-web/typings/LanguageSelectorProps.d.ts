/**
 * This file was generated from LanguageSelector.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix UI Content Team
 */
import { CSSProperties } from "react";
import { ListValue, ListExpressionValue } from "mendix";

export type PositionEnum = "left" | "right" | "top" | "bottom";

export type TriggerEnum = "click" | "hover";

export interface LanguageSelectorContainerProps {
    name: string;
    class: string;
    style?: CSSProperties;
    tabIndex?: number;
    languageOptions: ListValue;
    languageCaption: ListExpressionValue<string>;
    position: PositionEnum;
    trigger: TriggerEnum;
    hideForSingle: boolean;
}

export interface LanguageSelectorPreviewProps {
    className: string;
    style: string;
    styleObject?: CSSProperties;
    readOnly: boolean;
    languageOptions: {} | { type: string } | null;
    languageCaption: string;
    position: PositionEnum;
    trigger: TriggerEnum;
    hideForSingle: boolean;
}
