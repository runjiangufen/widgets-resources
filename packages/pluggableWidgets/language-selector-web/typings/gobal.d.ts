export interface MXSession {
    clearCachedSessionData();
    getUserObject(): MxObject;
}

export interface MXGlobalObject {
    reloadWithState();
    reload();
    session: MXSession;
    data: MxObject.Data;
}

declare global {
    interface Window {
        mx: MXGlobalObject;
    }
}
