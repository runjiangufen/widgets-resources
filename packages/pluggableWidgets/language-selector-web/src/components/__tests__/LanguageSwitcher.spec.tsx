import { render } from "@testing-library/react";
import { createElement } from "react";
import { PositionEnum, TriggerEnum } from "typings/LanguageSelectorProps";
import { LanguageSwitcher, LanguageSwitcherProps } from "../LanguageSwitcher";

let props: LanguageSwitcherProps = {
    preview: false,
    currentLanguage: undefined,
    languageList: [],
    position: "left" as PositionEnum,
    onSelect: jest.fn(),
    trigger: "click" as TriggerEnum,
    className: ""
};
const language = { _guid: "111", value: "En us" };

describe("Language switcher", () => {
    it("renders the structure with empty language list", () => {
        const { asFragment } = render(<LanguageSwitcher {...props} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders the structure with language list and selected default language", () => {
        props = { ...props, languageList: [language], currentLanguage: language };
        const { asFragment } = render(<LanguageSwitcher {...props} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
