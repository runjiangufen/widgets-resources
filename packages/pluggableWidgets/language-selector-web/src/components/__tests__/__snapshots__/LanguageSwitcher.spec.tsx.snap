// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Language switcher renders the structure with empty language list 1`] = `
<DocumentFragment>
  <div
    class="widget-language-selector popupmenu"
  >
    <div
      class="popupmenu-trigger popupmenu-trigger-alignment"
    >
      <span
        class="current-language-text"
      />
      <span
        aria-hidden="true"
        class="language-arrow"
      >
        <img
          alt=""
          class="arrow-down"
        />
      </span>
    </div>
    <div
      class="popupmenu-menu popupmenu-position-left"
    />
  </div>
</DocumentFragment>
`;

exports[`Language switcher renders the structure with language list and selected default language 1`] = `
<DocumentFragment>
  <div
    class="widget-language-selector popupmenu"
  >
    <div
      class="popupmenu-trigger popupmenu-trigger-alignment"
    >
      <span
        class="current-language-text"
      >
        En us
      </span>
      <span
        aria-hidden="true"
        class="language-arrow"
      >
        <img
          alt=""
          class="arrow-down"
        />
      </span>
    </div>
    <div
      class="popupmenu-menu popupmenu-position-left"
    >
      <div
        class="popupmenu-basic-item active"
      >
        En us
      </div>
    </div>
  </div>
</DocumentFragment>
`;
